src/client/rekor.ts

```diff
@@ -236,6 +236,25 @@ export class Rekor {
     const data = await response.json();
     return data;
   }
+
+  /**
+   * Search the Rekor logs for matching the given query.
+   * @param opts {SearchLogQuery} Query to search the Rekor log
+   * @returns {Promise<Entry[]>} List of matching entries
+   */
+  public async searchLog(opts: SearchLogQuery): Promise<Entry[]> {
+    const url = `${this.baseUrl}/api/v1/log/entries/retrieve`;
+
+    const response = await this.fetch(url, {
+      method: 'POST',
+      body: JSON.stringify(opts),
+      headers: { 'Content-Type': 'application/json' },
+    });
+    checkStatus(response);
+
+    const data = await response.json();
+    return data;
```

Comment:
Note that the `Entry` type doesn't quite match the response format that comes back from the API. Instead of returning the raw response you'll want to map the results through `entryFromResponse` so they get translated into the correct format.

The reason for the mismatch is to work around the weird format Rekor uses where the entry is keyed by UUID. Instead of:

```json
{
  "5c1419ace1915869eef4426701d0763232da44d453e31ffc06d60ea61de36e25": {
    "body": "abc",
    "integratedTime": 103930923,
    . . . 
}
```
We translate it to:
```json
{
  "uuid": "5c1419ace1915869eef4426701d0763232da44d453e31ffc06d60ea61de36e25",
  "body": "abc",
  "integratedTime": 103930923,
  . . .
}
```